hearth/spec/hearth/configuration_spec.rb

```diff
@@ -31,49 +24,34 @@ def self.defaults
       it 'uses the config resolver' do
         options = { simple_option: 'test' }
         expect(Hearth::Config::Resolver).to receive(:resolve)
-          .with(an_instance_of(config_class), options, config_class.defaults)
-
-        config_class.new(**options)
+          .with(
+            an_instance_of(config_class),
+            options,
+            { simple_option: ['default'] }
+          ).and_call_original
+
+        config = config_class.new(**options)
+        expect(config.simple_option).to eq('test')
       end
 
       it 'resolves correctly when not passed any options' do
         expect(Hearth::Config::Resolver).to receive(:resolve)
-          .with(an_instance_of(config_class), {}, config_class.defaults)
-
-        config_class.new
-      end
-
-      it 'raises on invalid types' do
-        expect do
-          config_class.new(simple_option: 1)
-        end.to raise_error(
-          ArgumentError,
-          'Expected config[:simple_option] to be in [String], got Integer.'
-        )
-      end
-
-      it 'raises on unknown options' do
```

Comment:
Why remove this test? Do we not validate this anymore?

Reply:
It's validated implicitly because of struct call, so I figure we don't need to test code we don't control.

Reply:
Was just thinking about if decide to change away form using a struct to implement and whether we consider raising on unknowns an important part of the interface or not.  I'd argue it is since we want to give users information about having potentially used the wrong options.